### ISSUE TITLE
Training result of UETSC dataset differs from the pertained model

### ISSUE THREAD

**Issue by jinseokbae** [Mathux/ACTOR#12]
Hi,
first of all, Thank you for releasing the nicely arranged code.
Released code really helped understanding of the model, and I appreciate kindly written instructions for the code.

However, I encountered undesirable result on UETSC dataset, which comes from the model  I trained from the scratch.
I followed instruction on the dataset, also did not modify training script.
`python -m src.train.train_cvae --modelname cvae_transformer_rc_rcxyz_kl --pose_rep rot6d --lambda_kl 1e-5 --jointstype vertices --batch_size 20 --num_frames 60 --num_layers 8 --lr 0.0001 --glob --translation --no-vertstrans --dataset uestc --num_epochs 1000 --snapshot 100 --folder exps/uestc`

I double-checked whether saved opt.yml file from the experiment is identical to the pretrained model, but I found that no configurations were different.
Also, I conducted the same experiment twice but results were identical.

This is the train log of the released pretrained model,
<img width="349" alt="스크린샷 2022-02-10 오전 10 41 38" src="https://user-images.githubusercontent.com/30554812/153321217-d0b267b9-6e8f-4b2f-985a-613f3be1a4fd.png">

and these are the train logs from the trainings I conducted.
<img width="370" alt="스크린샷 2022-02-10 오전 10 42 40" src="https://user-images.githubusercontent.com/30554812/153321296-bd182995-4196-4fad-9228-84f03af152e4.png">

As my training result on HumanAct dataset is same with the pre-trained model, I guess no training environmental issue involves here.

Is there any possible reason for this failure of the training?

**Comment by Mathux** [comment]
Hello [jinseokbae](https://github.com/jinseokbae),

This is indeed very strange. Did you download the already preprocessed data from this repo? 
I tried to reproduce your failure without any success. I just retrain the model using Pytorch 1.10, but it works fine. What version of pytorch/cuda are you using?

Did you try again and got the same results? 

Unfortunately, I am not sure of how I can help.

**Comment by jinseokbae** [comment]
Hi @Mathux,
I appreciate your reply.
I tried Pytorch 1.8.1 with CUDA version 10.1.
Also I downloaded preprocessed UESTC data as instruction guided.
If preprocessed dataset I downloaded is exactly identical to the dataset you used, Pytorch version might be an issue.
By the way, thank you for the current test on it :)

**Comment by yhZhai** [comment]
Hi @jinseokbae ,

I encountered the same problem as you. I ran 1500 epochs and the results I got is
`gen    | 103.23 +/- 2.59 | 95.37 +/- 0.79 | 73.2 +/- 0.27 | 27.38 +/- 0.43 | 14.04 +/- 0.07`
And the training loss I got is
|    kl   |  312.11390 |
|    rc   |   0.00760  |
|  rcxyz  |   0.00706  |
|  mixed  |   0.01778  |

I use PyTorch 1.11.0 with CUDA 11.6.
May I know your reproduced results?

Thanks!

**Comment by yhZhai** [comment]
> Hi @jinseokbae ,
> 
> I encountered the same problem as you. I ran 1500 epochs and the results I got is `gen | 103.23 +/- 2.59 | 95.37 +/- 0.79 | 73.2 +/- 0.27 | 27.38 +/- 0.43 | 14.04 +/- 0.07` And the training loss I got is | kl | 312.11390 | | rc | 0.00760 | | rcxyz | 0.00706 | | mixed | 0.01778 |
> 
> I use PyTorch 1.11.0 with CUDA 11.6. May I know your reproduced results?
> 
> Thanks!

After several trials, I solved this problem by increasing learning rate, training epochs and reducing the num_layers to 4. Hope this can solve your problem.
